openmdao/core/vec_wrapper.py

```diff
@@ -891,15 +885,26 @@ def _get_flattened_sizes(self):
         return [[(n, m['size']) for n, m in self._get_vecvars()
                     if m.get('owned')]]
 
-    def _apply_unit_derivatives(self, varnames):
+    def _apply_unit_derivatives(self):
         """ Applies derivative of the unit conversion factor to params
-        sitting in vector for names passed in varnames."""
+        sitting in vector.
+        """
         if self.deriv_units:
-            for name in varnames:
-                conv = self._vardict[name].get('unit_conv')
-                if conv is not None:
-                    self[name] *= conv[0]
-
+            for name, meta in iteritems(self._vardict):
+                if 'unit_conv' in meta:
+                    meta['val'] *= meta['unit_conv'][0]
+
+    # def _apply_units(self):
+    #     """ Applies the unit conversion factor to params
+    #     sitting in vector.
+    #     """
+    #     for name, meta in iteritems(self._vardict):
+    #         if 'unit_conv' in meta and 'owned' in meta:
+    #             scale, offset = meta['unit_conv']
+    #             val = meta['val']
+    #             if offset != 0.0:
+    #                 val += offset
+    #             val *= scale
```

Comment:
My hope was to completely eliminate unit conversion in the vecwrappers and handle it the way we handle it with derivatives now, but I wasn't able to get it working on the first try.  I haven't given up yet though, so I kept this code there to support that.